lib/format.js

```diff
@@ -0,0 +1,87 @@
+/** @babel */
+import {Range} from 'atom';
+import {rangeFromLineNumber} from 'atom-linter';
+import ruleURI from 'eslint-rule-documentation';
+
+// (editor: Object) => function
+export default function format(editor) {
+	const buffer = editor.getBuffer();
+	const filePath = editor.getPath();
+
+	// (report: Object) => Array<Object>
+	return report => {
+		if (!report) {
+			return [];
+		}
+
+		const [{messages}] = report.results;
+
+		return messages.map(message => {
+			return {
+				filePath,
+				fix: selectMessageFix(message, buffer),
+				html: selectMessageHtml(message),
+				range: selectMessageRange(editor, message),
+				type: selectMessageType(message)
+			};
+		});
+	};
+}
+
+// (message: Object, buffer: Object) => Array<Object>
+function selectMessageFix(message, buffer) {
+	const fix = {message};
+
+	if (!fix) {
+		return null;
+	}
+
+	const {
+		range: rawRange = [],
+		text: newText
+	} = fix;
+
+	const points = rawRange.map(index => buffer.positionForCharacterIndex(index));
+	const range = new Range(...points);
+	return {newText, range};
+}
+
+// (message: Object, err: Error) => string
+function selectMessageHtml(result) {
+	const {message, ruleId} = result;
+	const {url} = ruleURI(ruleId || '');
+
+	const link = ruleId ? `(<a href=${url}>${ruleId}</a>)` : '';
+	return `<span>${[message, link].filter(Boolean).join(' ')}</span>`;
+}
+
+// (editor: Object, message: Object) => Array<number>
+function selectMessageRange(editor, message) {
+	const {line, column} = message;
+	const endLine = typeof message.endLine === 'number' ? message.endLine : line;
+	const endColumn = typeof message.endColumn === 'number' ?
+		message.endColumn : column;
+
+	const start = [line, column]
+		.filter(coord => typeof coord === 'number')
+		.map(coord => coord - 1);
+
+	const end = [endLine, endColumn]
+		.filter(coord => typeof coord === 'number')
+		.map(coord => coord - 1);
+
+	if (end.length === 2) {
+		return [start, end];
+	}
+
+	try {
```

Comment:
This is never reached as `end.length` is always `2`.

Master:
<img width="151" alt="screen shot 2016-11-23 at 22 19 32" src="https://cloud.githubusercontent.com/assets/170270/20567500/79c85d1e-b1cb-11e6-8c75-8be8f67f3478.png">

This PR:
<img width="150" alt="screen shot 2016-11-23 at 22 20 01" src="https://cloud.githubusercontent.com/assets/170270/20567503/8035b3ea-b1cb-11e6-9215-80c12ade7664.png">

Notice the underline.